src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -540,9 +556,140 @@ private List<GeneratorTemplate> getModElementGeneratorTemplatesList(ModElement e
 			}
 		}
 
+		Objects.requireNonNull(getModElementListTemplates(element, performFSTasks, generatableElement)).forEach(gtl -> {
+			for (ListTemplate generatorTemplate : gtl.templates().keySet()) {
+				for (int i = 0; i < gtl.listData().size(); i++) {
+					if (gtl.templates().get(generatorTemplate).get(i) || !performFSTasks) {

```

Comment:
performFSTasks should not control if conditions are considered or not. Looking at this, seems if performFSTasks is set to false, all templates will be included, which is not how this should work.

Reply:
Seems I forgot to remove it when moving the code from another place.

Reply:
✔️ Fixed